_episodes/06-branching.md

```diff
@@ -0,0 +1,280 @@
+you continue to work on subject reconciliation. Once they are completed with
+their work, you can review their changes before merging their changes back into the master
+branch.
+
+When you create a branch, you are still going to have just one directory of
+files in your project. All of the files that you are working with are still
+going to be in same place. When you switch branches Git takes all of the files in your
+project directory and swaps them out so that they match what is on that branch.
+
+Let's look at an illustration.
+
+![Branching Illustrated](../fig/branching.gif)
+
+You are on your `master` branch, and have made several commits to it. Then you decide you
+want to test out your new reconciliation process. You create and switch to the
+`subject_recon` branch and commit your changes.
+
+These new commits are not on the `master` branch. Each time you switch between the
+branches, git switches out the files in your project directory. You can continue to make
+commits to `master`, but they will not appear in the `subject_recon` branch.
+
+If you are happy with your experiment, you can incorporate those changes into `master` by
+merging the `subject_recon` branch. During this process Git will create a new commit that
+merges the changes from the `subject_recon` branch. You can then delete the
+`subject_recon` branch, or continue to make changes, periodically merging them
+back in.
+
+You don't have to create branches from, or merge changes to, the `master` branch. You can
+have many branches and merge changes between those branches.
+
+## Viewing Branches
+
+Start by looking at current branches on local machine using the `git branch` command.
+
+~~~
+$ git branch
+* master
+~~~
+{: .bash}
+
+The asterisk tells you which branch you are on, also referred to as the branch that
+is checked out.
+
+## Creating Branches
+
+You create a branch using `git branch` followed by the name of the new branch.
+
+The branch name can branch whatever you want, but keep it simple: letters, numbers,
+dashes, and underscores. You cannot use spaces or special characters. Make the names
+useful and meaningful so you know what they refer to when you look at your list of
+branches. Many choose to name their branches with the issue number from their ticketing
+system.
+
+Now create a branch to begin working on your plan for metadata cleanup.
+
+~~~
+$ git branch cleanup-plan
+~~~
+{: .bash}
+
+When you hit return it will create the new branch.
+
+Let's take a look by using git branch again.
+
+~~~
+$ git branch
+  cleanup-plan
+* master
+~~~
+{: .bash}
+
+You can see by the asterisk that you are still on the master branch.
+
+## Switching Branches
+
+To switch to your new branch, you use the `git checkout` command.
+
+~~~
+$ git checkout cleanup-plan
+Switched to branch 'cleanup-plan'
+~~~
+{: .bash}
+
+You should see a confirmation message of the switch, but you can also confirm using git
+branch.
```

Comment:
Not sure if it's worth also noting that you can see in `git status`? Checking `git branch` keeps the flow going nicely though. If not here, it might be a good reminder to put in somewhere in the lesson.

Reply:
Yeah, it'd be a nice aside.

Reply:
added it!